### PR TITLE
Add cpu tests

### PR DESCRIPTION
**Pull request by maxjeblick**
This PR adds cpu-only integration tests for causal language problem type.
It uses a modified tiny llama model (https://huggingface.co/MaxJeblick/llama2-0b-unit-test). 
All tests run in ~3:30 minutes (exlucing setup).

I've removed `assert os.path.exists(os.path.join(cfg["output_directory"], "adapter_model.bin"))`, seems as if this assertion no longer works.

After this PR has been merged we can add more tests (and models) for the other problem types.

### CODE REVIEW

File: tests/integration/test_integration.py [hunk 1]
```diff
@@ -50,50 +50,61 @@ def get_experiment_status(path: str) -> str:
...
-
     train_path = os.path.join(tmp_path, "train_full.pq")
-
-    config_path = (
-        Path.cwd() / "tests" / "src" / "integration" / f"{config_name}.yaml"
-    ).resolve()
-    with open(config_path, "r") as fp:
+    with open(
+        os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), f"{config_name}.yaml"
```
Comment (pascal-pfeiffer) [inline tests/integration/test_integration.py:85]: Is there a reason to have unit tests in `src` and the integration tests in the parent dir?

**Review by pascal-pfeiffer — COMMENTED** [review]
Few comments, otherwise looks great! Thank you!

File: tests/integration/test_causal_language_modeling_oasst_cpu_cfg.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  mask_prompt_labels: true
+  parent_id_column: None
+  personalize: false
+  prompt_column:
+  - instruction
+  system_column: None
+  text_answer_separator: <|answer|>
+  text_prompt_start: <|prompt|>
+  text_system_start: <|system|>
+  train_dataframe: /tmp/pytest-of-max/pytest-1/test_oasst_training_cpu_BLEU_t0/train_full.pq
```
[the remark this answers is not in the document]
Comment (maxjeblick) [reply tests/integration/test_causal_language_modeling_oasst_cpu_cfg.yaml]: yes, makes total sense.

File: tests/integration/test_integration.py (hunk 1 above)
Comment (maxjeblick) [reply tests/integration/test_integration.py:85]: The idea to move integration tests out of src was to make them more explicit and have tests in src be unit tests for the respective src folder.

**Review by pascal-pfeiffer — APPROVED** [review]
lgtm!

**Pull request merged**